03-wordemb-pytorch/wordemb-skip.py

```diff
@@ -0,0 +1,122 @@
+from collections import defaultdict
+import math
+import time
+import random
+import torch
+import torch.nn.functional as F
+
+
+class Skip(torch.nn.Module):
+    def __init__(self, nwords, emb_size):
+        super(Skip, self).__init__()
+
+        """ word embeddings """
+        self.word_embedding = torch.nn.Embedding(nwords, emb_size)
+        """ context embeddings"""
+        self.context_embedding = torch.nn.Embedding(nwords, emb_size)
+
+    def forward(self, word_pos, context_pos):
```

Comment:
@MysteryVaibhav: I had a look at the skipgram model, I think (I might be wrong) you are making a mistake in when you are calculating `logsigmoid`. The goal here is to make the current word predict its context, and when we need to make sure that the _probability_ (or log of probability) of these context words are high. But, in your implementation, you are optimizing the unnormalized score of the context without looking at other words in the vocabulary. 

https://github.com/neubig/nn4nlp-code/blob/6ae371f9cdf2dc939ed8da1bd5aa8ca8457206c7/03-wordemb-pytorch/wordemb-skip.py#L22-L28

Specifically, In line 27, I think you are only taking a log-sigmoid over the target index rather than the whole vocabulary. 

Does this make sense?

Reply:
@danishpruthi : As discussed with you earlier, this was just the word2vec impl. of minimizing the cosine distance b/w a word and its context word (without the negative sampling part). I have changed it to match the exact dynet implementation which involves taking a softmax over the context words.